src/lib/hooks/useMinimalShellMode.tsx

```diff
@@ -50,7 +51,7 @@ export function useMinimalShellMode() {
         })
       }
     })
-  }, [minimalShellInterp, store.shell.minimalShellMode])
+  }, [minimalShellInterp, minimalShellMode])
```

Comment:
This seems like a regression — we're triggering re-render when the shell mode changes. We really shouldn't. However, it's not a new regression. It looks like we regressed on this in https://github.com/bluesky-social/social-app/commit/8e9cf182c2e247203b6b5ea9ae701c039945d6a0#r131878136. The original fix was https://github.com/bluesky-social/social-app/pull/1691. We'd need a different fix for this, but since it's already broken, I guess it doesn't block this PR.

Reply:
Let's try to fix it while we're here. How can we react to the state change without passing in `minimalShellMode` as the dep?

Reply:
I think we’d need to do an explicit subscription — ie hold an array of listeners, and invoke them instead of having it be a state variable. I’d prefer to decouple that though since it’s a bit more involved.

Reply:
Tbh the actual thing i wanted to do is just to make it a “shared value” from Reanimated. So no setState — maybe a set function that toggles it. We’d do this for all gesture/animation-driving things.

Reply:
We should also change it from a boolean to a float and have it be driven by scroll so that it scrolls away and back “with” the content. Don’t have to do now ofc

Reply:
Okay sg